### PR TITLE
Bool literal

### PR DESCRIPTION
**Pull request by waywardmonkeys**
A couple of additional static analyzer fixes.

### CODE REVIEW

File: imgui.cpp [hunk 1]
```diff
@@ -11578,15 +11578,15 @@ void ImGui::ColorEditOptionsPopup(const float* col, ImGuiColorEditFlags flags)
...
+        if (RadioButton("HSV", opts & ImGuiColorEditFlags_HSV)) opts = (opts & ~ImGuiColorEditFlags__InputsMask) | ImGuiColorEditFlags_HSV;
+        if (RadioButton("HEX", opts & ImGuiColorEditFlags_HEX)) opts = (opts & ~ImGuiColorEditFlags__InputsMask) | ImGuiColorEditFlags_HEX;
     }
     if (allow_opt_datatype)
     {
         if (allow_opt_inputs) Separator();
-        if (RadioButton("0..255",     (opts & ImGuiColorEditFlags_Uint8) ? 1 : 0)) opts = (opts & ~ImGuiColorEditFlags__DataTypeMask) | ImGuiColorEditFlags_Uint8;
-        if (RadioButton("0.00..1.00", (opts & ImGuiColorEditFlags_Float) ? 1 : 0)) opts = (opts & ~ImGuiColorEditFlags__DataTypeMask) | ImGuiColorEditFlags_Float;
+        if (RadioButton("0..255",     opts & ImGuiColorEditFlags_Uint8)) opts = (opts & ~ImGuiColorEditFlags__DataTypeMask) | ImGuiColorEditFlags_Uint8;
+        if (RadioButton("0.00..1.00", opts & ImGuiColorEditFlags_Float)) opts = (opts & ~ImGuiColorEditFlags__DataTypeMask) | ImGuiColorEditFlags_Float;
```
Comment (ocornut) [inline imgui.cpp]: Hmm, wouldn't implicit int to bool conversion warn on some more zealous compiler settings? I have a feeling it would?

Comment (waywardmonkeys) [reply imgui.cpp]: Good question / point ... maybe an explicit cast here would be best. I just didn't want to change it to `... ? true : false`.

**Comment by waywardmonkeys** [comment]
I will revise this when I wake up in the morning!

**Comment by waywardmonkeys** [comment]
Updated and force pushed.

**Comment by ocornut** [comment]
Thank you Bruce, squashed and merged. If you have more in the future let's try to merge them together.

**Pull request merged**